Drop unusued optional dependencies, so that they're not found by mistake
Make other optional dependencies that are used 'required'
Check assimp version

--- CMakeLists.txt~	2016-05-27 14:31:10.000000000 +0200
+++ CMakeLists.txt	2016-06-01 14:30:28.696448120 +0200
@@ -63,12 +63,19 @@
 # ----------------------------------------------------
 ADD_REQUIRED_DEPENDENCY("eigen3 >= 3.0.5")
 ADD_OPTIONAL_DEPENDENCY("eigenpy >= 1.3.0")
-ADD_OPTIONAL_DEPENDENCY("metapod >= 1.0.7")
-ADD_OPTIONAL_DEPENDENCY("urdfdom >= 0.2.10")
-ADD_OPTIONAL_DEPENDENCY("lua5.1")
+#ADD_OPTIONAL_DEPENDENCY("metapod >= 1.0.7")
+ADD_REQUIRED_DEPENDENCY("urdfdom >= 0.2.10")
+#ADD_OPTIONAL_DEPENDENCY("lua5.1")
 ADD_OPTIONAL_DEPENDENCY("hpp-fcl >= 0.4.1")
 IF(HPP_FCL_FOUND AND URDFDOM_FOUND)
-  ADD_REQUIRED_DEPENDENCY("assimp >= 3.0")
+  ADD_REQUIRED_DEPENDENCY("assimp >= 2.0")
+if (NOT ${ASSIMP_VERSION} VERSION_LESS "2.0.1150")
+  add_definitions(-DASSIMP_UNIFIED_HEADER_NAMES)
+  message(STATUS "Assimp version has unified headers")
+else()
+  message(STATUS "Assimp version does not have unified headers")
+endif()
+
 ENDIF(HPP_FCL_FOUND AND URDFDOM_FOUND)
 
 SET(BOOST_COMPONENTS filesystem unit_test_framework system)
